src/bot.ts

```diff
@@ -136,6 +138,34 @@ export class DiscordBot {
         return this.provisioner;
     }
 
+    public lockChannel(channel: Discord.GuildChannel) {
+        const key = `${channel.guild.id}|${channel.id}`;
+        if (this.channelLocks[key]) {
+            return;
+        }
+        let i: NodeJS.Timeout;
+        const p = new Promise((resolve) => {
+            i = setInterval(resolve, this.config.limits.discordSendDelay);
+            this.channelLocks[key] = {i, p};
```

Comment:
shouldn't this be outside the promise?

Reply:
Nope, because we need to set the interval pointer *after* it's been set, and `setInterval` needs the resolve function exposed.